### PR TITLE
Fixes #21127,  latex printer fails on printing a singularity function raised to a power

### LINKED ISSUE #21127: _print_SingularityFunction() got an unexpected keyword argument 'exp'
On a Jupyter Notebook cell, type the following:

```python
from sympy import *
from sympy.physics.continuum_mechanics import Beam
# Young's modulus
E = symbols("E")
# length of the beam
L = symbols("L")
# concentrated load at the end tip of the beam
F = symbols("F")
# square cross section
B, H = symbols("B, H")
I = B * H**3 / 12
# numerical values (material: steel)
d = {B: 1e-02, H: 1e-02, E: 210e09, L: 0.2, F: 100}

b2 = Beam(L, E, I)
b2.apply_load(-F, L / 2, -1)
b2.apply_support(0, "fixed")
R0, M0 = symbols("R_0, M_0")
b2.solve_for_reaction_loads(R0, M0)
```

Then:

```
b2.shear_force()
```

The following error appears:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/usr/local/lib/python3.8/dist-packages/IPython/core/formatters.py in __call__(self, obj)
    343             method = get_real_method(obj, self.print_method)
    344             if method is not None:
--> 345                 return method()
    346             return None
    347         else:

/usr/local/lib/python3.8/dist-packages/sympy/interactive/printing.py in _print_latex_png(o)
    184         """
    185         if _can_print(o):
--> 186             s = latex(o, mode=latex_mode, **settings)
    187             if latex_mode == 'plain':
    188                 s = '$\\displaystyle %s$' % s

/usr/local/lib/python3.8/dist-packages/sympy/printing/printer.py in __call__(self, *args, **kwargs)
    371 
    372     def __call__(self, *args, **kwargs):
--> 373         return self.__wrapped__(*args, **kwargs)
    374 
    375     @property

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in latex(expr, **settings)
   2913 
   2914     """
-> 2915     return LatexPrinter(settings).doprint(expr)
   2916 
   2917 

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in doprint(self, expr)
    252 
    253     def doprint(self, expr):
--> 254         tex = Printer.doprint(self, expr)
    255 
    256         if self._settings['mode'] == 'plain':

/usr/local/lib/python3.8/dist-packages/sympy/printing/printer.py in doprint(self, expr)
    289     def doprint(self, expr):
    290         """Returns printer's representation for expr (as a string)"""
--> 291         return self._str(self._print(expr))
    292 
    293     def _print(self, expr, **kwargs):

/usr/local/lib/python3.8/dist-packages/sympy/printing/printer.py in _print(self, expr, **kwargs)
    327                 printmethod = '_print_' + cls.__name__
    328                 if hasattr(self, printmethod):
--> 329                     return getattr(self, printmethod)(expr, **kwargs)
    330             # Unknown object, fall back to the emptyPrinter.
    331             return self.emptyPrinter(expr)

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in _print_Add(self, expr, order)
    381             else:
    382                 tex += " + "
--> 383             term_tex = self._print(term)
    384             if self._needs_add_brackets(term):
    385                 term_tex = r"\left(%s\right)" % term_tex

/usr/local/lib/python3.8/dist-packages/sympy/printing/printer.py in _print(self, expr, **kwargs)
    327                 printmethod = '_print_' + cls.__name__
    328                 if hasattr(self, printmethod):
--> 329                     return getattr(self, printmethod)(expr, **kwargs)
    330             # Unknown object, fall back to the emptyPrinter.
    331             return self.emptyPrinter(expr)

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in _print_Mul(self, expr)
    565             # use the original expression here, since fraction() may have
    566             # altered it when producing numer and denom
--> 567             tex += convert(expr)
    568 
    569         else:

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in convert(expr)
    517                                isinstance(x.base, Quantity)))
    518 
--> 519                 return convert_args(args)
    520 
    521         def convert_args(args):

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in convert_args(args)
    523 
    524                 for i, term in enumerate(args):
--> 525                     term_tex = self._print(term)
    526 
    527                     if self._needs_mul_brackets(term, first=(i == 0),

/usr/local/lib/python3.8/dist-packages/sympy/printing/printer.py in _print(self, expr, **kwargs)
    327                 printmethod = '_print_' + cls.__name__
    328                 if hasattr(self, printmethod):
--> 329                     return getattr(self, printmethod)(expr, **kwargs)
    330             # Unknown object, fall back to the emptyPrinter.
    331             return self.emptyPrinter(expr)

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in _print_Add(self, expr, order)
    381             else:
    382                 tex += " + "
--> 383             term_tex = self._print(term)
    384             if self._needs_add_brackets(term):
    385                 term_tex = r"\left(%s\right)" % term_tex

/usr/local/lib/python3.8/dist-packages/sympy/printing/printer.py in _print(self, expr, **kwargs)
    327                 printmethod = '_print_' + cls.__name__
    328                 if hasattr(self, printmethod):
--> 329                     return getattr(self, printmethod)(expr, **kwargs)
    330             # Unknown object, fall back to the emptyPrinter.
    331             return self.emptyPrinter(expr)

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in _print_Mul(self, expr)
    569         else:
    570             snumer = convert(numer)
--> 571             sdenom = convert(denom)
    572             ldenom = len(sdenom.split())
    573             ratio = self._settings['long_frac_ratio']

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in convert(expr)
    505         def convert(expr):
    506             if not expr.is_Mul:
--> 507                 return str(self._print(expr))
    508             else:
    509                 if self.order not in ('old', 'none'):

/usr/local/lib/python3.8/dist-packages/sympy/printing/printer.py in _print(self, expr, **kwargs)
    327                 printmethod = '_print_' + cls.__name__
    328                 if hasattr(self, printmethod):
--> 329                     return getattr(self, printmethod)(expr, **kwargs)
    330             # Unknown object, fall back to the emptyPrinter.
    331             return self.emptyPrinter(expr)

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in _print_Add(self, expr, order)
    381             else:
    382                 tex += " + "
--> 383             term_tex = self._print(term)
    384             if self._needs_add_brackets(term):
    385                 term_tex = r"\left(%s\right)" % term_tex

/usr/local/lib/python3.8/dist-packages/sympy/printing/printer.py in _print(self, expr, **kwargs)
    327                 printmethod = '_print_' + cls.__name__
    328                 if hasattr(self, printmethod):
--> 329                     return getattr(self, printmethod)(expr, **kwargs)
    330             # Unknown object, fall back to the emptyPrinter.
    331             return self.emptyPrinter(expr)

/usr/local/lib/python3.8/dist-packages/sympy/printing/latex.py in _print_Pow(self, expr)
    649         else:
    650             if expr.base.is_Function:
--> 651                 return self._print(expr.base, exp=self._print(expr.exp))
    652             else:
    653                 tex = r"%s^{%s}"

/usr/local/lib/python3.8/dist-packages/sympy/printing/printer.py in _print(self, expr, **kwargs)
    327                 printmethod = '_print_' + cls.__name__
    328                 if hasattr(self, printmethod):
--> 329                     return getattr(self, printmethod)(expr, **kwargs)
    330             # Unknown object, fall back to the emptyPrinter.
    331             return self.emptyPrinter(expr)

TypeError: _print_SingularityFunction() got an unexpected keyword argument 'exp'
```

### PR DESCRIPTION
**Pull request by ayushbisht2001**
I have tried to fix the issue #21127.
#### References to other Issues or PRs

Fixes  #21127

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by moorepants** [comment]
You'll need to add a unit test that checks this.

File: sympy/printing/printer.py [hunk 1]
```diff
@@ -326,7 +326,7 @@ def _print(self, expr, **kwargs):
             for cls in classes:
                 printmethod = '_print_' + cls.__name__
                 if hasattr(self, printmethod):
-                    return getattr(self, printmethod)(expr, **kwargs)
```
Comment (ayushbisht2001) [inline sympy/printing/printer.py:329]: @moorepants, why this change is failing in some checks? Have I done it falsely?

File: sympy/printing/tests/test_latex.py [hunk 2]
```diff
@@ -213,6 +213,8 @@ def test_latex_SingularityFunction():
         r"{\left\langle x - 4 \right\rangle}^{-2}"
     assert latex(SingularityFunction(x, 4, -1)) == \
         r"{\left\langle x - 4 \right\rangle}^{-1}"
+    assert latex(SingularityFunction(x, 0, 0)**2) == \
+        r"{\\left\\langle x \\right\\rangle}^{0}"
```
Comment (oscargus) [inline sympy/printing/tests/test_latex.py:217]: The r in front of the string means that you do not have to escape `\`, so either remove the r or replace `\\` with `\`.

File: sympy/printing/tests/test_latex.py [hunk 3]
```diff
@@ -70,10 +70,10 @@ def _latex(self, printer):
             return "foo(%s)" % printer._print(self.args[0])
     assert latex(R(x)) == r"foo(x)"
 
-    class R(Abs):
+    class B(Abs):
```
Comment (oscargus) [inline sympy/printing/tests/test_latex.py:73]: Why this change? Anything called R elsewhere? (Doesn't look lite it.)

File: sympy/printing/printer.py (hunk 1 above)
Comment (oscargus) [reply sympy/printing/printer.py:329]: `**kwargs` must be there since they control the exact behavior of the printing. So this must be restored.

Probably the failing tests are due to this as it changes the printing of (more or less) *all* functions. 

You should approach the method `_print_SingularityFunction` to introduce the changes.

**Comment by oscargus** [comment]
I close this as it was fixed in the already merged #21171
